server/src/main/java/org/elasticsearch/transport/RemoteClusterAware.java

```diff
@@ -74,7 +75,10 @@ protected Map<String, List<String>> groupClusterIndices(Set<String> remoteCluste
         Map<String, List<String>> perClusterIndices = new HashMap<>();
         Set<String> clustersToRemove = new HashSet<>();
         for (String index : requestIndices) {
-            int i = index.indexOf(RemoteClusterService.REMOTE_CLUSTER_INDEX_SEPARATOR);
+            // ensure that `index` is a remote name and not a datemath expression which includes ':' symbol
+            // since datemath expression after evaluation should not contain ':' symbol
```

Comment:
Hmm this raises questions about the interaction between cross-cluster search and date-math expressions that I do not feel qualified to answer. @javanna perhaps you have opinions about this? Are these things supposed to work together?

Reply:
Currently this code trips over if search is queried with complex date-math index name
```
curl -X GET "localhost:9200/%3Cdatemath-%7B2001-01-01-13%7C%7C%2B1h%2Fh%7Byyyy-MM-dd-HH%7C-07%3A00%7D%7D%3E/_count?error_trace"
```

{"error":{"root_cause":[{"type":"no_such_remote_cluster_exception","reason":"no such remote cluster: [<datemath-{2001-01-01-13||+1h/h{yyyy-MM-dd-HH|-07]","stack_trace":"org.elasticsearch.transport.NoSuchRemoteClusterException: no such remote cluster: [<datemath-{2001-01-01-13||+1h/h{yyyy-MM-dd-HH|-07]\n\tat org.elasticsearch.server@8.12.0-SNAPSHOT/org.elasticsearch.cluster.metadata.ClusterNameExpressionResolver.resolveClusterNames(ClusterNameExpressionResolver.java:41)\n\tat org.elasticsearch.server@8.12.0-SNAPSHOT/org.elasticsearch.transport.RemoteClusterAware.groupClusterIndices(RemoteClusterAware.java:85)\n\tat org.elasticsearch.server@8.12.0-SNAPSHOT/org.elasticsearch.transport.RemoteClusterService.groupIndices(RemoteClusterService.java:181)

Reply:
I am afraid that a local date math expression containing `:` will be mistaken for a cross-cluster search, like raised above. That happens even if you escape the `:` (just tested it). On the contrary, I'd expect a cross-cluster search request including a date math expressions with a `:` to work ok.

I would recommend opening a separate PR to fix this shortcoming. Strange that it has not been reported so far, it probably means that date math expression with a `:` in them are not widely used, as they effectively don't work in Elasticsearch for a while, at least since we removed support for indices containing `:` in their names.

Reply:
OK, I'm moving out logic which handles complex datemath expressions to separate PR